cmd/main.go

```diff
@@ -167,26 +167,45 @@ func main() {
 func leaseHandler(podName string, mc controller.ModifyController, leaseChannel chan *v1.Lease) {
 	var cancel context.CancelFunc = nil
 
-	for lease := range leaseChannel {
-		currentLeader := *lease.Spec.HolderIdentity
-
-		klog.V(6).InfoS("leaseHandler: Lease updated", "currentLeader", currentLeader, "podName", podName)
-
-		if currentLeader == podName && cancel == nil {
-			var ctx context.Context
-			ctx, cancel = context.WithCancel(context.Background())
-			klog.InfoS("leaseHandler: Starting ModifyController", "podName", podName, "currentLeader", currentLeader)
-			go mc.Run(*workers, ctx)
-		} else if currentLeader != podName && cancel != nil {
-			klog.InfoS("leaseHandler: Stopping ModifyController", "podName", podName, "currentLeader", currentLeader)
-			cancel()
-			cancel = nil
-		}
-	}
+	klog.InfoS("leaseHandler: Looking for external-resizer lease holder")
+
+	timer := time.NewTimer(*resyncPeriod)
+	defer timer.Stop()
+
+	for {
+		select {
+		case lease, ok := <-leaseChannel:
+			if !ok {
```

Comment:
Is this `if` statement (or reading `ok` at all) necessary? As far as I can tell it should be impossible for `leaseChannel` to close

Reply:
Discussed offline; check will remain primarily for unit testing purposes and it is also generally a good practice.